forge-game/src/main/java/forge/game/card/Card.java

```diff
@@ -197,6 +197,8 @@ public class Card extends GameEntity implements Comparable<Card>, IHasSVars {
     private boolean tokenCard = false;
     private Card copiedPermanent;
     private boolean copiedSpell = false;
+    private boolean hideChosenCard = false;
```

Comment:
this seems quite GUI specific

It would be cleaner if Card class only has fields for game rules stuff as much as possible

There should be no need for these anyway?

Reply:
I'm afraid that you're right – there is probably a better way to do this.

The problem is that some of these Alchemy cards have a player select a card in hand without revealing it, then the engine keeps track of the perpetual effect applied to that card. Forge would always detail the chosen card on the Perpetual Effect, basically revealing hand information which seemed wrong.

There has to be a solution, just need to think on it a bit...